pandas/core/dtypes/cast.py

```diff
@@ -904,15 +904,19 @@ def maybe_infer_to_datetimelike(value, convert_dates=False):
     def try_datetime(v):
         # safe coerce to datetime64
         try:
-            v = tslib.array_to_datetime(v, errors='raise')
+            # GH19671
+            v = tslib.array_to_datetime(v,
+                                        require_iso8601=True,
+                                        errors='raise')
         except ValueError:
 
             # we might have a sequence of the same-datetimes with tz's
             # if so coerce to a DatetimeIndex; if they are not the same,
             # then these stay as object dtype
             try:
                 from pandas import to_datetime
-                return to_datetime(v)
+                # GH19671
+                return to_datetime(v, require_iso8601=True)
```

Comment:
So I mean to change this with `tslibs.conversion.datetime_to_datetime64(v)` (which is what would be used in `to_datetime` under the hood when `tslib.array_to_datetime` failed.

Reply:
changes implemented. 👍